src/script/bitcoinconsensus.h

```diff
@@ -54,7 +56,7 @@ enum
 /// txTo correctly spends the scriptPubKey pointed to by scriptPubKey under
 /// the additional constraints specified by flags.
 /// If not NULL, err will contain an error/success code for the operation
-EXPORT_SYMBOL int bitcoinconsensus_verify_script(const unsigned char *scriptPubKey, unsigned int scriptPubKeyLen,
+EXPORT_SYMBOL int bitcoinconsensus_verify_script(const unsigned char *scriptPubKey, unsigned int scriptPubKeyLen, uint64_t amount,
```

Comment:
@jtimon Feel free to suggest what you think is right.

My proposal is this:
* The old version remains usable, with unmodified behaviour, and does not accept a bitcoinconsensus_SCRIPT_FLAGS_VERIFY_WITNESS.
* A new function is added, which does require the amount being passed, and does accept that flag.